src/v0/functions.ts

```diff
@@ -0,0 +1,250 @@
+      }
+    })
+  }
+
+  create(fn: Function): Promise<FunctionResponse> {
+    return new Promise(async (resolve, reject) => {
+      const uri = `${this.options.base}${this.endpoint}/${this.options.user}`
+      try {
+        const response = await this.http.getClient().post(uri, fn)
+
+        return resolve({
+          data: response.data.results[0] as Function,
+          metadata: { count: response.data.count }
+        } as FunctionResponse)
+      } catch (error) {
+        return reject(new FunctionCreationFailed(undefined, { error }))
+      }
+    })
+  }
+
+  delete(functionId: string): Promise<FunctionResponse> {
+    return new Promise(async (resolve, reject) => {
+      const uri = `${this.options.base}${this.endpoint}/${this.options.user}/${functionId}`
+      try {
+        const response = await this.http.getClient().delete(uri)
+        response.status !== 200 && reject(new FunctionDeleteFailed())
+
+        return resolve({
+          msg: response.data.msg
+        } as FunctionResponse)
+      } catch (err) {
+        return reject(new FunctionDeleteFailed())
+      }
+    })
+  }
+}
+
+export class FunctionsFetchFailed extends BaseError {
+  public name = 'FunctionsFetchFailed'
+  constructor(public message: string = 'Could not fetch functions', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionFetchFailed extends BaseError {
+  public name = 'FunctionFetchFailed'
+  constructor(public message: string = 'Could not fetch function', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionContentFetchFailed extends BaseError {
+  public name = 'FunctionContentFetchFailed'
+  constructor(public message: string = 'Could not fetch function content', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionPatchFailed extends BaseError {
+  public name = 'FunctionPatchFailed'
+  constructor(public message: string = 'Could not alter function', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionCreationFailed extends BaseError {
+  public name = 'FunctionCreationFailed'
+  constructor(public message: string = 'Could not create function', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionsCountFailed extends BaseError {
+  public name = 'FunctionsCountFailed'
+  constructor(public message: string = 'Could not count the functions', properties?: any) {
+    super(message, properties)
+  }
+}
+
+export class FunctionDeleteFailed extends BaseError {
+  public name = 'FunctionDeleteFailed'
+  constructor(public message: string = 'Could not delete function', properties?: any) {
+    super(message, properties)
+  }
+}
+export class FunctionBindingFailed extends BaseError {
+  public name = 'FunctionBindingFailed'
+  constructor(public message: string = 'Could not bind function', properties?: any) {
+    super(message, properties)
+  }
+}
```

Comment:
Some of the error classes don't seem to be used. Is that also intentional?

Reply:
ah good catch, yeah I'll remove the unused ones.